File: Estrutura De Dados/Aluno.py
class Aluno:
    def __init__(self, matricula, nome):
        self.matricula = matricula
        self.nome = nome

    def __str__(self):
        return f"Aluno [matricula={self.matricula}, nome={self.nome}]"

    def __eq__(self, other):
        if not isinstance(other, Aluno):
            return False
        return self.matricula == other.matricula

    def __hash__(self):
        return hash(self.matricula)

if __name__ == "__main__":
    HashAluno = set()
    HashAluno.add(Aluno(1, "pedro"))

    for item in HashAluno:
        print(item)

    obj = Aluno(1, "carla")

    if obj in HashAluno:
        print("Encontrado")
    else:
        print("Não encontrado")

File: Estrutura De Dados/Aluno.java
package EstruturaDeDados;

public class Aluno {
    int matricula;
    String nome;

    public Aluno(int matricula, String nome) {
        this.matricula = matricula;
        this.nome = nome;
    }

    public int getMatricula() {
        return matricula;
    }
    public void setMatricula(int matricula) {
        this.matricula = matricula;
    }
    public String getNome() {
        return nome;
    }
    public void setNome(String nome) {
        this.nome = nome;
    }

    @Override
    public String toString() {
        return "Aluno [matricula=" + matricula + ", nome=" + nome + "]";
    }

    @Override
    public int hashCode() {
        final int prime = 31;
        int result = 1;
        result = prime * result + matricula;
        return result;
    }

    
    @Override
    public boolean equals(Object obj) {
        if (this == obj)
            return true;
        if (obj == null)
            return false;
        if (getClass() != obj.getClass())
            return false;
        Aluno other = (Aluno) obj;
        if (matricula != other.matricula)
            return false;
        return true;
    }
    
    

}
